pkg/controllers/plugin/util.go

```diff
@@ -4,16 +4,91 @@
 package plugin
 
 import (
+	"context"
+	"encoding/json"
 	"fmt"
 
 	"github.com/pkg/errors"
+	appsv1 "k8s.io/api/apps/v1"
+	batchv1 "k8s.io/api/batch/v1"
 	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
 
 	greenhouseapis "github.com/cloudoperators/greenhouse/pkg/apis"
+	greenhousev1alpha1 "github.com/cloudoperators/greenhouse/pkg/apis/greenhouse/v1alpha1"
+	"github.com/cloudoperators/greenhouse/pkg/clientutil"
 )
 
+// initClientGetter returns a RestClientGetter for the given Plugin.
+// If the Plugin has a clusterName set, the RestClientGetter is initialized from the cluster secret.
+// Otherwise, the RestClientGetter is initialized with in-cluster config
+func initClientGetter(
+	ctx context.Context,
+	client client.Client,
+	kubeClientOpts []clientutil.KubeClientOption,
+	plugin greenhousev1alpha1.Plugin,
+	pluginStatus greenhousev1alpha1.PluginStatus,
+) (
+	clusterAccessReadyCondition greenhousev1alpha1.Condition,
+	restClientGetter genericclioptions.RESTClientGetter,
+) {
+
+	clusterAccessReadyCondition = *pluginStatus.GetConditionByType(greenhousev1alpha1.ClusterAccessReadyCondition)
```

Comment:
I would suggest always setting the initial `clusterAccessReadyCondition` to unknown

```suggestion
	clusterAccessReadyCondition = greenhousev1alpha1.UnknownCondition(greenhousev1alpha1.ClusterAccessReadyCondition, "", "")
```

Reply:
I'm calling this function:
https://github.com/cloudoperators/greenhouse/blob/8bb9e5fa4b54a8672459502e6c659374d2098625/pkg/controllers/plugin/plugin_workload_status_controller.go#L94
which is already doing that here:
https://github.com/cloudoperators/greenhouse/blob/8bb9e5fa4b54a8672459502e6c659374d2098625/pkg/controllers/plugin/helm_controller.go#L159-L160